File: RemoveDuplicatesFromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int j = 0;
        for (int i = 1; i < nums.length; i++) {
            nums[j] = nums[i-1];
            if (nums[i] == nums[i-1]) {
                continue;
            }
            j++;
            nums[j] = nums[i];
        }
        return j+1;
    }
}

File: RemoveDuplicatesFromSortedArray.scala
object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
        if (nums.length == 0) {
            return nums.length;
        }
        var j:Int = 0;
        for (i <- 1 until nums.length) {
            if (nums(i) != nums(j)) {
                j += 1
                nums(j) = nums(i)
            }
        }
        j+1;
    }
}
